app/hooks/useQuestionStateInUrl.ts

```diff
@@ -83,17 +86,111 @@ export default function useQuestionStateInUrl(minLogo: boolean, initialQuestions
     setStateString(null)
   }
 
+  const moveToTop = (currentState: string, {pageid}: Question) => {
+    const removePageRe = new RegExp(`${pageid}.`, 'g')
+    window.scrollTo({
+      top: 0,
+      behavior: 'smooth',
+    })
+    return `${pageid}-${currentState.replace(removePageRe, '')}`
+  }
+
+  /*
+   * Get all related questions of the provided `question` that aren't already displayed on the site
+   */
+  const unshownRelatedQuestions = (
+    questions: Question[],
+    questionProps: Question
+  ): RelatedQuestions => {
+    const {relatedQuestions} = questionProps
+
+    const onSiteAnswers = onSiteAnswersRef.current
+    const onSiteSet = new Set(onSiteAnswers.map(({pageid}) => pageid))
+
+    return relatedQuestions.filter((question) => {
+      const isOnSite = onSiteSet.has(question.pageid)
+      // hide already displayed questions, detect duplicates by pageid (pageid can be different due to redirects)
+      // TODO: #25 relocate already displayed to slide in as a new related one
+      const isAlreadyDisplayed = questions.some(({pageid}) => pageid === question.pageid)
+      return isOnSite && !isAlreadyDisplayed
+    })
+  }
+
+  /*
+   * Open the given question and add its subquestions - this will return an appropriate URL param string
+   */
+  const insertIntoState = (
+    state: string,
+    pageid: PageId,
+    relatedQuestions: RelatedQuestions
+  ): string =>
+    getStateEntries(state)
+      .map(([k, v]) => {
+        if (k === pageid.toString()) {
+          const newValue: QuestionState =
+            v === QuestionState.OPEN ? QuestionState.COLLAPSED : QuestionState.OPEN
+          const related = relatedQuestions
+            .filter((i) => i)
+            .map((r) => `${r.pageid}${QuestionState.SUBQUESTION}`)
+            .join('')
+          return `${k}${newValue}${related}`
+        }
+        return `${k}${v}`
+      })
+      .join('')
+
+  /*
+   * Update the window.location with the new URL state
+   */
+  const updateStateString = useCallback(
+    (newState: string) => {
+      const newSearchParams = new URLSearchParams(remixSearchParams)
+      newSearchParams.set('state', newState)
+      history.replaceState(newState, '', '?' + newSearchParams.toString())
+      setStateString(newState)
+    },
+    [remixSearchParams]
+  )
+
+  /*
+   * Add the given `questions` to the global questions map. This will not update the URL
+   */
+  const mergeNewQuestions = useCallback((questions: Question[]) => {
+    setQuestionMap((currentMap) =>
+      questions.reduce((map, question) => updateQuestionMap(question, map), new Map(currentMap))
+    )
+  }, [])
```

Comment:
the previous functions were simply extracted from `toggleQuestion` so they could be used elsewhere